cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -157,6 +158,9 @@ public void initialize(TwillContext context) {
       hConf = new Configuration();
       hConf.clear();
       hConf.addResource(new File(cmdLine.getOptionValue(RunnableOptions.HADOOP_CONF_FILE)).toURI().toURL());
+      // don't have tephra retry in order to give CDAP more control over when to retry and how.
+      hConf.set(TxConstants.Service.CFG_DATA_TX_CLIENT_RETRY_STRATEGY, "n-times");
+      hConf.setInt(TxConstants.Service.CFG_DATA_TX_CLIENT_ATTEMPTS, 0);
```

Comment:
Wanted to keep this PR about how programs interact with the transaction service. Making sure our system services do the right thing is a bigger change.